packages/web3/src/v3/SpaceRegistrar.ts

```diff
@@ -41,13 +41,30 @@ export class SpaceRegistrar {
     }
 
     public getSpace(spaceId: string): Space | undefined {
-        if (this.spaces[spaceId] === undefined) {
-            const spaceAddress = SpaceAddressFromSpaceId(spaceId)
-            if (!spaceAddress || spaceAddress === ethers.constants.AddressZero) {
-                return undefined // space is not found
+        // aellis 10/2024 we don't really need to cache spaces, but they instantiate a lot of objects
+        // for the contracts and it's worth not wasting memory if we need to access the same space multiple times
+        // this code is also used on the server so we don't want to cache spaces for too long
+        const space = this.spaces[spaceId]?.space || this.createSpace(spaceId)
+        this.pruneSpaces()
+        this.spaces[spaceId] = { lastSeen: new Date().getTime(), space }
+        return space
+    }
+
+    private pruneSpaces(): void {
+        // clear out spaces that haven't been seen in 5 seconds
+        const fiveSecondsAgo = new Date().getTime() - 1000 * 5
+        for (const spaceId in this.spaces) {
+            if (this.spaces[spaceId].lastSeen < fiveSecondsAgo) {
+                delete this.spaces[spaceId]
```

Comment:
@texuf this turns each `get` operation from O(1) to O(n) - though n approaches to zero every 5 seconds which is okay i guess. still, i think we'd be better off using something like this: https://www.npmjs.com/package/lru-cache

Reply:
that's a very good point...

Reply:
fun tidbit from the documentation:
```
If you truly wish to use a cache that is bound only by TTL expiration, consider using a Map object, and calling setTimeout to delete entries when they expire. It will perform much better than an LRU cache.
```
there's even an example

Reply:
ha! i like that a lot

Reply:
@texuf though technically we need to bound primarily by space and not necessarily by time. as long as the cache size does not exceed a certain limit i wouldn't care when the cache gets cleared but i would definitely care about not exceeding a max space limit regardless of ttl

Reply:
that's a good point. LRU would be better in that regard.

Reply:
updated to lru-cache